docs/src/main/asciidoc/mongodb.adoc

```diff
@@ -573,6 +573,11 @@ a discriminator field inside the document. It can be useful when working with ab
 
 Quarkus will automatically register all the classes annotated with `@BsonDiscriminator` with the POJO codec.
 
+The POJO Codec have enhanced generic support via `PropertyCodecProvider`,
+Quarkus will automatically register any `PropertyCodecProvider` with the POJO Codec.
+If you deploy your application as a native executable, be aware that reflection may not work properly with generic types, so you may need to register it manually for reflection,
```

Comment:
Why can't we do it automatically in the extension? I'm not sure I understand what the problem is?

Reply:
It's a general comment on generic usage, we cannot know which type will be used so if no code use it in a way that we can discover it (for example, this type is not an entity, not returned by JAX-RS, ...) it may be missing on the native image.

This sentence comes from some discussion around generic type usage, I don't remember it precisely

Reply:
@gsmet are you OK with this comment or do you want me to change it ?

Reply:
I still don't understand what the problem is. Could you give an example of what is supposedly not working?

Reply:
If you use an `Optional<T>` as a field of a of `Pojo` class and use a `MySubPojo` class at runtime as `T` in a way that can be discovered by GraalVM (for example, don't return it from a JAX-RS endpoint or stuff like this), you may need to register it for reflection.

Reply:
I can remove this sentence if you prefere